fastcrypto/src/traits.rs

```diff
@@ -219,7 +219,25 @@ pub trait RecoverableSigner {
     type Sig: RecoverableSignature<Signer = Self, PubKey = Self::PubKey>;
 
     /// Sign as a recoverable signature.
-    fn sign_recoverable(&self, msg: &[u8]) -> Self::Sig;
+    fn try_sign_recoverable(&self, msg: &[u8]) -> Result<Self::Sig, FastCryptoError>;
```

Comment:
when can this function fail? (or, why is it try_?)

Reply:
I was just wondering that myself. The answer is that it can't fail and that the function should just be sign_recoverable. The reason the error was there in the first place was to handle the situations where k = 0 or s = 0 during the ecdsa signing, both which have negligible probability since we use RFC6979 to generate the k.